src/dist/signatures.rs

```diff
@@ -0,0 +1,51 @@
+//! Signature verification support for Rustup.
+
+// TODO: Determine whether we want external keyring support
+
+use pgp::types::KeyTrait;
+use pgp::{Deserializable, SignedPublicKey, StandaloneSignature};
+
+use crate::errors::*;
+
+// const SIGNING_KEY_BYTES: &[u8] = include_bytes!("rust-signing-key.asc");
+const SIGNING_KEY_BYTES: &[u8] = include_bytes!("../../tests/mock/signing-key.pub.asc");
+
+lazy_static::lazy_static! {
+    static ref SIGNING_KEYS: Vec<SignedPublicKey> = {
+        pgp::SignedPublicKey::from_armor_many(std::io::Cursor::new(SIGNING_KEY_BYTES))
+            .map_err(squish_internal_err).unwrap()
+            .0
+            .collect::<std::result::Result<Vec<_>, _>>()
+            .map_err(squish_internal_err).unwrap()
+    };
+}
+
+fn squish_internal_err<E: std::fmt::Display>(err: E) -> Error {
+    ErrorKind::SignatureVerificationInternalError(format!("{}", err)).into()
+}
+
+pub fn verify_signature(content: &str, signature: &str) -> Result<bool> {
```

Comment:
Could we please change this so that `content` is given as a `&dyn Read` rather than `&str` ?  I'm thinking ahead to verifying signatures on updates to rustup.

Reply:
sure